homeassistant/components/yeelight/services.yaml

```
@@ -1,96 +1,190 @@
 set_mode:
+  name: Set mode
   description: Set a operation mode.
+  target:
+    entity:
+      integration: yeelight
+      domain: light
   fields:
-    entity_id:
-      description: Name of the light entity.
-      example: "light.yeelight"
     mode:
-      description: Operation mode. Valid values are 'last', 'normal', 'rgb', 'hsv', 'color_flow', 'moonlight'.
-      example: "moonlight"
+      name: Mode
+      description: Operation mode.
+      required: true
+      selector:
+        select:
+          options:
+            - 'color_flow'
+            - 'hsv'
+            - 'last'
+            - 'moonlight'
+            - 'normal'
+            - 'rgb'
+
 set_color_scene:
+  name: Set color scene
   description: Changes the light to the specified RGB color and brightness. If the light is off, it will be turned on.
+  target:
+    entity:
+      integration: yeelight
+      domain: light
   fields:
-    entity_id:
-      description: Name of the light entity.
-      example: "light.yeelight"
     rgb_color:
+      name: RGB color
       description: Color for the light in RGB-format.
       example: "[255, 100, 100]"
+      selector:
+        object:
     brightness:
-      description: The brightness value to set (1-100).
-      example: 50
+      name: Brightness
+      description: The brightness value to set.
+      selector:
+        number:
+          min: 0
+          max: 100
 set_hsv_scene:
+  name: Set HSV scene
   description: Changes the light to the specified HSV color and brightness. If the light is off, it will be turned on.
+  target:
+    entity:
+      integration: yeelight
+      domain: light
   fields:
-    entity_id:
-      description: Name of the light entity.
-      example: "light.yeelight"
     hs_color:
+      name: Hue/sat color
       description: Color for the light in hue/sat format. Hue is 0-359 and Sat is 0-100.
       example: "[300, 70]"
+      selector:
+        object:
     brightness:
-      description: The brightness value to set (1-100).
-      example: 50
+      name: Brightness
+      description: The brightness value to set.
+      selector:
+        number:
+          min: 0
+          max: 100
 set_color_temp_scene:
+  name: Set color temperature scene
   description: Changes the light to the specified color temperature. If the light is off, it will be turned on.
+  target:
+    entity:
+      integration: yeelight
+      domain: light
   fields:
-    entity_id:
-      description: Name of the light entity.
-      example: "light.yeelight"
     kelvin:
+      name: Kelvin
       description: Color temperature for the light in Kelvin.
       example: 4000
+      selector:
+        number:
+          min: 0
+          max: 1000000
```

This value is clamped in this range
```suggestion
          min: 1700
          max: 6500
```

---

```suggestion
          min: 1700
          max: 6500
          step: 100
          unit_of_measurement: K
          mode: slider
```